src/Raven.Server/Documents/Handlers/VersioningHandler.cs

```diff
@@ -143,5 +140,45 @@ private Task GetRevisions()
 
             return Task.CompletedTask;
         }
+
+        [RavenAction("/databases/*/revisions/zombied", "GET", "/databases/*/revisions/zombied?etag={long.MaxValue}&pageSize=25")]
+        public Task GetZombiedRevisions()
+        {
+            var versioningStorage = Database.DocumentsStorage.VersioningStorage;
+            if (versioningStorage.Configuration == null)
+                throw new VersioningDisabledException();
+
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var etag = GetLongQueryString("etag");
+                var pageSize = GetPageSize(); // TODO: limit page size. By default it is int.MaxValue!
+                var revisions = versioningStorage.GetZombiedRevisions(context, etag, pageSize).ToArray();
```

Comment:
Why are you forcing us to materialize it all at one shot? There is no need to do that.
ETag check can be done in the same way we do it for the `/docs` endpoint.